Change linker to trim everything by default in 7.0

agocke opened the pull request:
The new trimmer default will be to trim everything, which will be represented
by a new TrimMode called 'full'. The previous behavior will be TrimMode
'partial'. Other options for TrimMode, and the TrimmerDefaultAction property,
are deprecated.

All scenarios where trimming is enabled by default (Blazor WASM, Xamarin) will
be expected to opt-in to the previous behavior using `TrimMode=partial`. Console
apps will use the new `Full` behavior in net7.0, unless otherwise specified.

agocke committed:
Update linker to trim everything by default

The new trimmer default will be to trim everything, which will be represented
by a new TrimMode called 'full'. The previous behavior will be TrimMode
'partial'. Other options for TrimMode, and the TrimmerDefaultAction property,
are deprecated.

agocke commented:
Counterpart to https://github.com/dotnet/linker/pull/2856

sbomer reviewed (approved):
LGTM, just a few nits. Thanks a lot!

agocke committed:
Set TrimMode partial for Asp.NET

agocke committed:
Fix whitespace

agocke committed:
Respond to PR comments

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/linker build 20220707.1

Microsoft.NET.ILLink.Analyzers , Microsoft.NET.ILLink.Tasks
 From Version 7.0.100-1.22354.1 -> To Version 7.0.100-1.22357.1

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/linker build 20220707.2

Microsoft.NET.ILLink.Analyzers , Microsoft.NET.ILLink.Tasks
 From Version 7.0.100-1.22354.1 -> To Version 7.0.100-1.22357.2

agocke committed:
Merge remote-tracking branch 'upstream/main' into trimmode

agocke committed:
Merge remote-tracking branch 'upstream/darc-main-a7abfb98-e20d-49e1-8e9d-95393ec16cfa' into trimmode

agocke commented:
Just updated to bring in changes from https://github.com/dotnet/sdk/pull/26472

agocke committed:
Merge remote-tracking branch 'upstream/main' into trimmode

agocke committed:
Change ReferencedExe test to account for aggressive trimming

agocke committed:
Merge remote-tracking branch 'upstream/main' into trimmode
